### ISSUE TITLE
Report on Incremental Build

### ISSUE THREAD

**Issue by RonJeffries** [jekyll/jekyll#6912]
- [X] I believe this to be a bugXXX concern, not a question about using Jekyll.
- [X] I updated to the latest Jekyll (or) if on GitHub Pages to the latest `github-pages`
- [X] This is a feature request or background info.
- [X] I am on (or have tested on) ***macOS*** 10+

---

Enough with the questions. :) Here's my report. I run my web site, ronjeffries.com, using jekyll 2.7.something. It would have been nice to have incremental build, as it would be faster and simplify uploading new material. My pal and I DL'd and installed the latest Jekyll and learned this.

First, render_all was missing. We replaced some calls to render_all with super and things seemed to be OK. Not terribly important except that we could find almost no information on line but for one obscure location that suggested the super.

More important is that Jekyll ran continuously. We quickly isolated this to the fact that my installation has a plug-in that we wrote which generates a number of category indexes, by going through all posts and making lists of all those in each category and creating a page for that category.

As is probably obvious, creating a page during Jekyll's running convinced it that a save had taken place, and therefore it regenerated. (At least that's what we think triggered it.)

What is somewhat interesting is that in version 2, creating those pages doesn't trigger a build, even when I'm running jekyll serve, which does rebuild whenever I actually save a file. So I'm not certain it was the creation of the category indexes that triggered the loop, I just think it's the way to bet.

Anyway, I doubt I'm the only person who creates pages on the fly. If I am, well, no problem, I'm good with 2.7.x. What strikes me as a possible improvement would be some information in the config listing paths and file names that should NOT trigger a rebuild even if they change. This would certainly fix my problem if it is what I believe it is.

This is just for your info and consideration, in hopes that it'll be useful. It's not a priority for me and I'm happy as a clam, assuming clams are pretty happy. Thanks!

**Comment by DirtyF** [comment]
Related: https://github.com/jekyll/jekyll/issues/6418

**Comment by pathawks** [comment]
There's not much that can be done with the information provided. There is no `2.7.x` version of Jekyll, [`2.5.3` was the last `2.x` version](https://jekyllrb.com/docs/history/#v2-5-3). If you could use something like `git bisect` to narrow down when this behavior was changed, it would make it easier to figure out what is happening.

Jekyll does not (should not?) watch the `_site/` directory for changes that would trigger incremental build.

I strongly agree that plugins should have better access to incremental regeneration data :+1:

> If I am, well, no problem, I'm good with 2.7.x.

**This** is one of the most underrated advantages of using a static site generator :beers:

**Comment by DirtyF** [comment]
> my plan is just to live without incremental, and without upgrading. perfectly happy with that.

That's a pretty good plan.

Nevertheless, I'm curious if there any "technical debt"  preventing you from [upgrading to v3.x](https://jekyllrb.com/docs/upgrading/2-to-3/) 😉

**Comment by RonJeffries** [comment]
Main reason not to upgrade is that it's working and I was happy to have my build frozen. I thought incremental build might be useful. When we installed 3.7x, some of my plugins that called render_all didn't work and I THINK but am mot sure that our "super" call actually fixed that. This was part of our code that handled categories and tags as we needed. (Recall that Jekyll used to be very opinionated about blog format, and my site isn't arranged quite that way.) So I'm not at all sure that it renders correctly even without the looping, and since I'm satisfied with what I've got, no real incentive to upgrade.
